Tasks/Sepseleva/Task7/bl.py

```diff
@@ -0,0 +1,64 @@
+import data 
+from prettytable import PrettyTable
+
+def search_by_type(type, value):
+    res = data.search_by_type(type, value, l=[])
+    t = show_contents(res)
+    if t:
+       return f"""
+        ***All the results of your search '{type}: {value} '***
+{t}"""
+    return f'There are no such books in our store'
+    #Used formatting like that first: 
+    # f"Category: {category}\n\n"+"\n".join([f"{i['name']}, {i['author']} (price - ${i['price']:.2f}, quantity in stock - {i['number']})" for i in res]) 
+    #Then decided to use prettytable, seems to me it looks a bit nicer
+
+def add_basket(name, number):
+    try:
+        validate_number(number)
+    except ValueError:
+        return "The quantity of books should be a positive number"
+    try:
+        res = data.add_basket(name, number)
+    except RuntimeError as er:
+        return er
+    if res:
+        return "The items have been added to the basket"
+    return "There's no such a book in our store"
+    
+def show_basket():
+    res = data.show_basket()
+    t = show_contents(res)
+    if t:
+        return f"""
+        ***Items in your basket***
+{t}"""
+    return "Your basket is empty now. To add items input '2'"
```

Comment:
do not provide a user with UI suggestions from bl, these two layers should know nothing about each other

Reply:
ThankQ, I got it)

Reply:
ThanQ, I got it)